templates/content-certificate.php

```diff
@@ -38,19 +37,31 @@
 	</div>
 </div>
 
-<div class="llms-print-certificate no-print" id="llms-print-certificate">
-	<button class="llms-button-secondary" onClick="window.print()" type="button">
-		<?php echo _e( 'Print', 'lifterlms' ); ?>
-		<i class="fa fa-print" aria-hidden="true"></i>
-	</button>
+<?php
+if ( $cert->can_user_manage() ) {
+	$is_sharing_allowed = $cert->is_sharing_allowed();
+	?>
+	<div class="llms-print-certificate no-print" id="llms-print-certificate">
+		<button class="llms-button-secondary" onClick="window.print()" type="button">
+			<?php echo _e( 'Print', 'lifterlms' ); ?>
+			<i class="fa fa-print" aria-hidden="true"></i>
+		</button>
 
-	<form action="" method="POST">
-		<button class="llms-button-secondary" type="submit" name="llms_generate_cert">
+		<form action="" method="POST">
+			<button class="llms-button-secondary" type="submit" name="llms_generate_cert">
 			<?php echo _e( 'Save', 'lifterlms' ); ?>
-			<i class="fa fa-cloud-download" aria-hidden="true"></i>
-		</button>
+				<i class="fa fa-cloud-download" aria-hidden="true"></i>
+			</button>
 
-		<input type="hidden" name="certificate_id" value="<?php echo get_the_ID(); ?>">
-		<?php wp_nonce_field( 'llms-cert-actions', '_llms_cert_actions_nonce' ); ?>
-	</form>
-</div>
+			<button class="llms-button-secondary" type="submit" name="<?php echo $is_sharing_allowed ? 'llms_disable_cert_sharing' : 'llms_enable_cert_sharing'; ?>">
+		  <?php echo ( $is_sharing_allowed ? _e( 'Disable sharing', 'lifterlms' ) : _e( 'Enable sharing', 'lifterlms' ) ); ?>
+				<i class="fa fa-share-alt" aria-hidden="true"></i>
+			</button>
```

Comment:
This template is shared both by `llms_my_certificate` and `llms_certificate` post types.
The first one is the post type to model the certificate that a user, who earned it, can see in his (e.g.) `www.example.what/dashboard/my-certificate/cert-title-{N}`, while the latter is basically used by _admins_ to view the certificate created at `www.example.what/certificate/cert-title`.

This button should be available only when viewing a `llms_my_certificate` post type.

@thomasplevy I was thinking that a new method could be added to the `LLMS_User_Certificate` class, along with the new `is_sharing_allowed()`, that could be called `is_sharing_possible()` and it could look like this:
```php
	/**
	 * Is sharing possible
	 *
	 * @since [version]
	 *
	 * @return bool
	 */
	public function is_sharing_possible() {
		/**
		 * Filter whether or not sharing is possible for a certificate
		 *
		 * @since [version]
		 *
		 * @param boolean               $possible    Whether or not sharing is possible.
		 * @param LLMS_User_Certificate $certificate Certificate class instance.
		 */
		return apply_filters( 'llms_certificate_is_sharing_possible', get_post_type( $this->get( 'id' ) ) === $this->db_post_type, $this );
	}
```

so that in the template we'll have:
```suggestion
			<?php if ( $cert->is_sharing_possible() ) : ?>
				<button class="llms-button-secondary" type="submit" name="<?php echo $is_sharing_allowed ? 'llms_disable_cert_sharing' : 'llms_enable_cert_sharing'; ?>">
					<?php echo ( $is_sharing_allowed ? _e( 'Disable sharing', 'lifterlms' ) : _e( 'Enable sharing', 'lifterlms' ) ); ?>
					<i class="fa fa-share-alt" aria-hidden="true"></i>
				</button>
			<?php endif; ?>
```

Reply:
@eri-trabiccolo You are right, but i think it's good idea for admin to see this button in preview like other buttons (Save, Print)

but we can make it useless, i mean if he clicked on it, it will not make anything, What do you think ?

Reply:
@alaa-alshamy I see what you mean, but honestly don't like the idea of having a button in that view that would do nothing, but let's wait for @thomasplevy opinion.

Reply:
A button that doesn't work leads to support requests and angry, frustrated, consfused, etc... users.

I don't think we need another method here though. I think we can just run a check against the post type in the template.

Reply:
You're right, I actually didn't want to create a method - it doesn't.really make a lot of sense having it - although I didn't want to have the logic in the template. But yeah let's go with the check in the template.

Reply:
I think this is an acceptable amount of "business logic" in a template.

Reply:
@thomasplevy @eri-trabiccolo Ok great, i will do that check